examples/gdp/ConstrainedLayout.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import division
+import pyomo.environ
+from pyomo.core import *
+from pyomo.gdp import *
+
+# 2-D Constrained layout example from http://minlp.org/library/lib.php?lib=GDP
+
+# Given a set of boxes and a set of circular areas, we determine a layout for
+# the boxes which minimizes the distance among them (according to various costs
+# associated with each pair of boxes) while ensuring that each box is within at
+# least one of the circular areas
+
+model = AbstractModel()
+
+# Have to specify M since this is nonlinear
+model.BigM = Suffix(direction=Suffix.LOCAL)
+model.BigM[None] = 7000
+
+model.BOXES = Set()
+model.CIRCLES = Set()
+
+model.Height = Param(model.BOXES)
+model.Length = Param(model.BOXES)
+
+model.Radius = Param(model.CIRCLES)
+model.CenterXCoord = Param(model.CIRCLES)
+model.CenterYCoord = Param(model.CIRCLES)
+
+# generate the list of possible pairs of boxes
+def box_pairs_filter(model, i, j):
+    return i < j
+model.BoxPairs = Set(initialize=model.BOXES * model.BOXES, dimen=2,
+                     filter=box_pairs_filter)
+
+# bounds for box coordinates
+# TODO: What are they trying to accomplish here? Why is it always index
+# 2 for the first two??
```

Comment:
This is an error in the original GAMS file. The intent is to find the min and max x and y coordinates that the boxes can have, based on the circle locations and radii. The code as written works because circle 2 has the higher x and y coordinate values.